### ISSUE TITLE
[JENKINS-23364] Rename hudson-dev:run to jenkins-dev:run

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#23175]
<p>To run Jenkins in development mode, you currently type:</p>

<p>    mvn hudson-dev:run</p>

<p>This command should probably be updated to:</p>

<p>    mvn jenkins-dev:run</p>

<p>See discussion here: <a href="https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4" class="external-link" target="_blank" rel="nofollow noopener">https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4</a></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kevinburke.jpg" title="kevinburke's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kevinburke">kevinburke</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23364" target="_blank">Rename hudson-dev:run to jenkins-dev:run</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=albers">albers</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2019-11-12T15:06:59+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>To run Jenkins in development mode, you currently type:</p>

<p>    mvn hudson-dev:run</p>

<p>This command should probably be updated to:</p>

<p>    mvn jenkins-dev:run</p>

<p>See discussion here: <a href="https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4" class="external-link" target="_blank" rel="nofollow noopener">https://groups.google.com/forum/#!topic/jenkinsci-dev/UNW6bWjcgF4</a></p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=203572&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-203572">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><tt>mvn jenkins-dev:run</tt> ist available only after applying special maven configuration while <tt>mvn hudson-dev</tt> works out of the box.</p>

<p>This behavior should be reversed.</p>
</pre>
</details>

</ul>
</details>

<p><tt>mvn jenkins-dev:run</tt> ist available only after applying special maven configuration while <tt>mvn hudson-dev</tt> works out of the box.</p>

<p>This behavior should be reversed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=204356&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204356">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>created <a href="https://github.com/jenkinsci/maven-hudson-dev-plugin/pull/1" class="external-link" target="_blank" rel="nofollow noopener">PR #1</a></p>
</pre>
</details>

</ul>
</details>

<p>created <a href="https://github.com/jenkinsci/maven-hudson-dev-plugin/pull/1" class="external-link" target="_blank" rel="nofollow noopener">PR #1</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/albers.png" title="albers's avatar" /> albers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-23364?focusedId=204367&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-204367">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The proposed change only covers interactive invocation from the command line (<tt>mvn jenkins-dev:run</tt> vs <tt>mvn hudson-dev:run</tt>). It has no impact on how the goals are called from existing build scripts (configuring goal exectuion in certain phases in pom.xml).</p>

<p>As far as I know, there is no way of parameterising the prefix. There can only be one.</p>

<p>In order to support both prefixes, the plugin has to be renamed back to <em>maven-hudson-dev-plugin</em> and a <em>goalPrefix</em> of <em>jenkins-dev</em> has to be configured in the plugin. But this would impact existing builds.</p>
</pre>
</details>

</ul>
</details>

<p>The proposed change only covers interactive invocation from the command line (<tt>mvn jenkins-dev:run</tt> vs <tt>mvn hudson-dev:run</tt>). It has no impact on how the goals are called from existing build scripts (configuring goal exectuion in certain phases in pom.xml).</p>

<p>As far as I know, there is no way of parameterising the prefix. There can only be one.</p>

<p>In order to support both prefixes, the plugin has to be renamed back to <em>maven-hudson-dev-plugin</em> and a <em>goalPrefix</em> of <em>jenkins-dev</em> has to be configured in the plugin. But this would impact existing builds.</p>
